File: solutions/c/two-fer/1/src/two_fer.c
#include "two_fer.h"
#include <stddef.h>
#include <string.h>

void two_fer(char *buffer, const char *name)
{
    buffer[0] = '\0';
    if(name != NULL && strlen(name) > 0)
    {
        strcat(buffer, "One for ");
        strcat(buffer, name);
        strcat(buffer, ", one for me.");
        return;
    }
    strcat(buffer, "One for you, one for me.");
}

File: solutions/go/two-fer/2/two_fer.go
// Package twofer provides a execution of twofer example.
package twofer

// TwoFer mix the name with another string.
func TwoFer(name string) string {
	if name == "" {
		name = "you"
	}

	return "One for " + name + ", one for me."
}
